src/components/Input.react.js

```diff
@@ -141,7 +141,10 @@ Input.propTypes = {
      * If true, changes to input will be sent back to the Dash server only on enter or when losing focus.
      * If it's false, it will sent the value back on every change.
      */
-    debounce: PropTypes.bool,
+    debounce: PropTypes.oneOfType([
+        PropTypes.string,
+        PropTypes.bool
+    ]),
```

Comment:
What's string for here? Describe it in the docstring?

Reply:
^^ applies to a bunch of these where either string or bool now accepts either... it's not clear to me which strings do what, or if they all evaluate to booleans or do something else entirely.

Reply:
this one was wrong actually - only `bool`. But for the other HTML boolean properties, HTML5 defines the only accepted values as the name of the attribute (case insensitive) or `""`: https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#boolean-attributes

That is, for `autoFocus`, the following assignments mean "turn on autofocus": `autoFocus=""`, `autoFocus="autoFocus"`, `autoFocus="autofocus"`, `autoFocus="AUTOFOCUS"`

And in order for it to be "turned off", it has to be omitted.

It seems like browsers are a little more loose and they evaluate these strings to be "If it's supplied, then it's applied" - it doesn't need to be the name of the attribute. This ends up being confusing because `checked="false"` is actually "turn on checked" because it's applied. MDN says:

> To be clear, the values "true" and "false" are not allowed on boolean attributes. To represent a false value, the attribute has to be omitted altogether. This restriction clears up some common misunderstandings: With checked="false" for example, the element’s checked attribute would be interpreted as true because the attribute is present.

So, we could either:
1. Just keep it as a string and allow users to navigate this spec. Move on for now and revist later.
2. Provide a preference but allow all strings: `autoFocus: oneOfType([oneOf(['autoFocus', 'autofocus', '']), propTypes.string])`
3. Be more opionated and only include the common forms like `'autoFocus': oneOf(['', 'autoFocus', 'autofocus', 'AUTOFOCUS'])` but omit the other permutations like `auToFoCus` that are technically valid.
3. Be even more opinionated and only include the most sane-looking version: `autoFocus='autoFocus'` and omit the other versions that are technically valid.

Reply:
I think I prefer 2 or 3

Reply:
So does `false` (not `'false'`) turn `autoFocus` off? Or is `None` -> `null` the only value that disables it?

If this check is only applied in dev mode, I'd vote to be more opinionated - (3) seems reasonable, plus `false` if that's allowed to disable. I'd omit `''` unless it's particularly common usage, as explicitly allowing a normally falsy value to mean true is bound to cause confusion. Either way we should document a recommended value, something like:
> autoFocus is enabled by `'autoFocus'`, but alternate capitalizations `'autofocus'` and `'AUTOFOCUS'` are also accepted.

Reply:
I thought that `false` ended up getting rendered as `'false'` looks like I'm wrong about that:
![image](https://user-images.githubusercontent.com/1280389/56099511-fd2dd980-5edb-11e9-8b0c-7634a8943bbe.png)

(https://codepen.io/chriddyp/pen/MREmpK?editors=0010)

Similarly in our toolchain:
```python
import dash
import dash_html_components as html
import dash_core_components as dcc

app = dash.Dash(__name__)
app.layout = html.Div([
    dcc.Input(id='1', readOnly=True),
    dcc.Input(id='2', readOnly=False),
    dcc.Input(id='3', readOnly='readonly'),
    dcc.Input(id='4', readOnly='readOnly'),
    dcc.Input(id='5', readOnly='true'),
    dcc.Input(id='6', readOnly='false'),
    dcc.Input(id='7', readOnly=''),
    dcc.Input(id='8', readOnly='""'),
])

if __name__ == '__main__':
    app.run_server(debug=True)
```

![image](https://user-images.githubusercontent.com/1280389/56099521-12a30380-5edc-11e9-91d1-5c77d1dba03a.png)

`#7` is pretty confusing - to get `""` into the DOM, you have to pass in `'""'` not `''`


So I suppose we could allow `boolean` for these properties as well.

Reply:
Oy, 7 is insane, definitely do not allow any version of that! So yeah, `true`, `false`, and a couple of capitalizations of `autoFocus`.

Reply:
a small section about the attributes in dash-docs might be necessary given all the experiments we have, as most of our audiences are not Frontend expert, we should align the preference in the demo code too